lib/scenic/schema_dumper.rb

```diff
@@ -3,6 +3,34 @@
 module Scenic
   # @api private
   module SchemaDumper
+    # A hash to do topological sort
+    class TSortableHash < Hash
+      include TSort
+
+      alias tsort_each_node each_key
+      def tsort_each_child(node, &block)
+        fetch(node).each(&block)
+      end
+    end
+
+    # Query for the dependencies between views
+    DEPENDENT_SQL = <<~SQL.freeze
+      SELECT distinct dependent_ns.nspname AS dependent_schema
+      , dependent_view.relname AS dependent_view
+      , source_ns.nspname AS source_schema
+      , source_table.relname AS source_table
+      FROM pg_depend
+      JOIN pg_rewrite ON pg_depend.objid = pg_rewrite.oid
+      JOIN pg_class as dependent_view ON pg_rewrite.ev_class = dependent_view.oid
+      JOIN pg_class as source_table ON pg_depend.refobjid = source_table.oid
+      JOIN pg_namespace dependent_ns ON dependent_ns.oid = dependent_view.relnamespace
+      JOIN pg_namespace source_ns ON source_ns.oid = source_table.relnamespace
+      WHERE dependent_ns.nspname = ANY (current_schemas(false)) AND source_ns.nspname = ANY (current_schemas(false))
+      AND source_table.relname != dependent_view.relname
+      AND source_table.relkind IN ('m', 'v') AND dependent_view.relkind IN ('m', 'v')
+      ORDER BY dependent_view.relname;
```

Comment:
```suggestion
      AND source_table.relkind IN ('m', 'v') AND dependent_view.relkind IN ('m', 'v');
```
Iterating over result does not need this order so this line can be removed.